lib/pbench/test/unit/server/test_server_configuration.py

```diff
@@ -0,0 +1,220 @@
+from http import HTTPStatus
+
+import pytest
+import requests
+
+
+class TestServerConfiguration:
+    @pytest.fixture()
+    def query_get(self, client, server_config):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+        """
+
+        def query_api(
+            key: str, expected_status: HTTPStatus = HTTPStatus.OK
+        ) -> requests.Response:
+            k = f"/{key}" if key else ""
+            response = client.get(f"{server_config.rest_uri}/server/configuration{k}")
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    @pytest.fixture()
+    def query_put(self, client, server_config, pbench_admin_token):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            pbench_admin_token: Provide an admin authentication token
+        """
+
+        def query_api(
+            key: str,
+            expected_status: HTTPStatus = HTTPStatus.OK,
+            token: str = pbench_admin_token,
+            **kwargs,
+        ) -> requests.Response:
+            k = f"/{key}" if key else ""
+            response = client.put(
+                f"{server_config.rest_uri}/server/configuration{k}",
+                headers={"authorization": f"bearer {token}"},
+                **kwargs,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def test_get_bad_keys(self, query_get):
+        response = query_get("xyzzy", HTTPStatus.BAD_REQUEST)
+        assert response.json == {
+            "message": "Unrecognized keyword ['xyzzy'] given for parameter key; allowed keywords are ['dataset-lifetime', 'server-banner', 'server-state']"
+        }
+
+    def test_get1(self, query_get):
+        response = query_get("dataset-lifetime")
+        assert response.json == {"dataset-lifetime": "3650"}
+
+    def test_get_all(self, query_get):
+        response = query_get(None, HTTPStatus.OK)
+        assert response.json == {
+            "dataset-lifetime": "3650",
+            "server-state": {"status": "enabled"},
+            "server-banner": None,
+        }
```

Comment:
For some definition of "now" we get a Flask 404 error because the trailing slash isn't mapped. While I didn't use your `key` parameter hack (though in retrospect I kinda like it), I "now" test this both ways and get identical results. 🕥